apps/fares/lib/highest_lowest_fare.ex

```diff
@@ -54,9 +55,20 @@ defmodule Fares.HighestLowestFare do
   def highest_fare(route, trip, origin_id, destination_id, fare_fn) do
     route
     |> get_fares(trip, origin_id, destination_id, fare_fn)
+    |> Enum.filter(fn fare -> fare.reduced == nil end)
     |> Enum.max_by(& &1.cents, fn -> nil end)
   end
 
+  def reduced_fare(route, trip, origin_id, destination_id, fare_fn \\ &Repo.all/1) do
```

Comment:
One small thing I missed in the first pass: could you please add a typespec for this new function?